controllers/ajax.py

```diff
@@ -408,13 +408,11 @@ def updatelastpage():
                    last_page_subchapter=lastPageSubchapter,
                    last_page_scroll_location=lastPageScrollLocation,
                    last_page_accessed_on=datetime.datetime.utcnow())
-        db.commit()
         db((db.user_sub_chapter_progress.user_id == auth.user.id) &
            (db.user_sub_chapter_progress.chapter_id == lastPageChapter) &
            (db.user_sub_chapter_progress.sub_chapter_id == lastPageSubchapter)).update(
                    status=completionFlag,
                    end_date=datetime.datetime.utcnow())
-        db.commit()
```

Comment:
These were added to try to reduce the number of deadlocks that were occurring in the database under load.  It seems to me that they had the desired effect.   How does this mess up testing?  We could take them back out and see if the deadlocks come back.

Reply:
When testing, the best approach IMHO is to start with a clean database in a known state. My new testing addition (in a PR to come) call and Ajax function, then do a `db.rollback()` to discard any changes and keep the database in this known state.

So, this could be worked around in testing, if it make the production environment better, would would be a bit ugly. It's your call. If you want to omit these changes, I'll update the PR.

Thanks for the rapid feedback. That helps a lot.

Reply:
Lets go ahead and take them out.

Reply:
If I start to see crashes there again, then we will figure out a new solution

Reply:
Sounds good. So, as I understand it, this PR needs no changes?